Client/src/components/NavBar.jsx

```diff
@@ -23,41 +22,41 @@ const NavBar = () => {
     setActive(e.target.innerText);
   };
 
-  // const logOut = async () => {
-  //   let myHeaders = new Headers();
-  //   myHeaders.append("Content-Type", "application/json");
-
-  //   let requestOptions = {
-  //     method: "POST",
-  //     headers: myHeaders,
-  //     redirect: "follow",
-  //     credentials: "include",
-  //   };
-
-  //   let user = cookies.get("user");
-
-  //   if (user) {
-  //     fetch(
-  //       "https://technoob-staging.azurewebsites.net/api/v1/authenticate/logout",
-  //       requestOptions
-  //     )
-  //       .then((response) => {
-  //         if (response.status === 200) {
-  //           setIsLoggedIn(false);
-  //           setUserProfile(null);
-  //           cookies.remove("user");
-  //         }
-  //         return response.json();
-  //       })
-  //       .catch((error) => {
-  //         console.log("error", error);
-  //       });
-  //   }
-  // };
-  // const handleClick = async () => {
-  //   await logOut();
-  //   navigate("/Home");
-  // };
+  const logOut = async () => {
+    let myHeaders = new Headers();
+    myHeaders.append("Content-Type", "application/json");
```

Comment:
Same as this too, all fetch calls should be replaced with serverApi calls.

Reply:
See this a s a reference.


`const login = async () => {
    let raw = JSON.stringify({
      password: user.Password,
      username: user.UserName,
    });
    const abortController = new AbortController();
    try {
      const response = await serverApi.post(
          '/authenticate/login',
          raw,
          {
            signal: abortController.signal,
            headers: {
              'content-type': 'application/json',
            }
          }
      )
       console.log(response);

       if (response.status === 200) {
         navigate("/dashboard");
         const responseData = response.data;
         const userInfo = {
           ...responseData.data,

    
          
            
    

          
          Expand Down
    
    
  
        };
        setUserProfile(userInfo.user);
        setIsLoggedIn(true);
        sessionStorage.setItem("userData", JSON.stringify(userInfo.user));
        sessionStorage.setItem("user_token", response.data.token);
        //localStorage.setItem("user", JSON.stringify(userInfo));
        if (userInfo.user.role === "admin") {
          setDashboardToggle({
            displayToggle: true,
            toggleValue: "User Dashboard",
          });
        }
      }
    
    } catch (error) {
      setUser({ error: error.response?.data?.message || "Failed, Please contact admin", UserName: "", Password: "" });
    }
    
  }`